queue_job/models/queue_job.py

```diff
@@ -14,9 +14,18 @@
 
 _logger = logging.getLogger(__name__)
 
+from odoo.models import MetaModel
+
 
 def channel_func_name(method):
-    return '<%s>.%s' % (method.im_class._name, method.__name__)
+    method_class = False
+    for key, value in method.func_globals.iteritems():
```

Comment:
You are right, the function is not yet bound to the class at this point. Your check does not seem to work better, in my case, the class is not in the globals of the function. It seems we can only check that at runtime when the delay is called.